rotkehlchen/accounting/events.py

```diff
@@ -247,15 +247,28 @@ def add_buy_and_corresponding_sell(
         - RemoteError if there is a problem reaching the price oracle server
         or with reading the response returned by the server
         """
+        # In the case where the fee asset is nominated in the asset that we receive
+        # take care that the fee is processed once we have the received asset in possesion.
+        if bought_asset != fee_currency:
```

Comment:
Hmm but if what you say is the case shouldn't it be:

`if bought_asset == fee_currency:`

So when the fee asset is the same as the asset we receive, we push the fee to the sell?

Reply:
Post mortem of my post mortem: This logic here is not needed since what you do first is to create the buy object. The else statement here was making sure that the fee was procesed in the buy part, so you were getting the buy and right after they fee substraction but for how the accounting is processed it doesn't matter for this case the where to put the fee since is always a buy before sell. 

So to keep it short: This `!=` was there to make sure that the fee was just after the acquisition but after reading it again is not needed because is always processed correctly. Just removed this logic as is not necessary. It is needed in the `add_sell_and_corresponding_buy`